libraries/abstractions/common_io/test/test_iot_flash.c

```diff
@@ -180,25 +293,16 @@ TEST_GROUP_RUNNER( TEST_IOT_FLASH )
  */
 TEST( TEST_IOT_FLASH, AFQP_IotFlashOpenClose )
 {
-    int32_t lRetVal;
     IotFlashHandle_t xFlashHandle;
+    int32_t lRetVal;
 
-    /* Open flash to initialize hardware. */
-    if( gIotFlashHandle == NULL )
```

Comment:
@vjjunnut, as most test also tear-down the handle on finish, this would only really apply to single test cases being run. Is this a common use-case in the test system and what is the gain from it? 

My understanding of the way the original tests was structured, running any group tests would result in the first test using the globally opened handle to then close it again and force the following tests to open it for them to succeed.

Reply:
Note that the caller can still control which handle is being opened as the refactoring also added a global "ltestIotFlashInstance" which can be set prior to running the tests.

Reply:
That is correct that the first test uses the global handle and then closes it so that the rest of the tests can open it again and close.
Usually there will be only one flash, so the instance will be the same. But even if there are multiple flash parts - we still need to be able to use it, even if it was open by the application which runs the tests. You can argue that application shall not open if its only to run the tests. But on some boards, flash is read first to get some context in startup code.

Reply:
That is correct that the first test uses the global handle and then closes it so that the rest of the tests can open it again and close.

Usually there will be only one flash, so the instance will be the same. But even if there are multiple flash parts - we still need to be able to use it, even if it was open by the application which runs the tests. You can argue that application shall not open if its only to run the tests. But on some boards, flash is read first to get some context in startup code.

Reply:
I fully understand that there might be a need to use the flash prior to running the tests. It just seems to me that it makes more sense to ensure that such tests setups leaves the hardware in a known state before running the tests. In other words, if the flash is used prior to the tests, the original owner using the flash should also release it.

Bringing in an external from outside the scope just to close it not only leaves the original owner being uncertain of the state of the handle, what happens if the one using the flash prior to the tests do not actually utilize the handle? 

Furthermore, none of the test cases as of now resets the global handle to NULL once closing it, which would let the following tests to believe that it was still valid, even while it was not.

Is there some reason that the original owner cannot release the handle prior to running the tests?

Reply:
@qiutongs Any thoughts on this?